dspace-uclouvain/src/main/java/org/dspace/uclouvain/core/utils/MetadataUtils.java

```diff
@@ -0,0 +1,99 @@
+    /** 
+    * Recovers the value for a specific field of a 'MetadataValue' list.
+    * 
+    * @param metadataValues: The list of 'MetadataValue' to search from.
+    * @param fieldId: The field to recover the value of.
+    * @return Value for the given metadata field (empty string if not found).
+    */
+    public String getMetadataFieldValueByFieldId(List<MetadataValue> metadataValues, String fieldId){
+        for (MetadataValue metadataValue: metadataValues) {
+            if (metadataValue.getMetadataField().toString().equals(fieldId)) {
+                return metadataValue.getValue();
+            }
+        }
+        return "";
+    }
+
+    /** 
+    * Recovers a list of all the values for a specific field of a 'MetadataValue' list.
+    * 
+    * @param metadataValues: The list of 'MetadataValue' to search from.
+    * @param fieldId: The field to recover values of.
+    * @return List of values for the given metadata field (empty list if no found).
+    */
+    public List<String> getAllMetadataFieldValuesByFieldId(List<MetadataValue> metadataValues, String fieldId){
+        List<String> values = new ArrayList<String>();
+        for (MetadataValue metadataValue: metadataValues) {
+            if (metadataValue.getMetadataField().toString().equals(fieldId)) {
+                values.add(metadataValue.getValue());
+            }
+        }
+        return values;
+    }
+
+    /** 
+    * This method is used to extract the real type of a DSpace item.
+    * 
+    * @param inputType: Type of a DSpace item of form "xx::xx::xx"
+    * @return The last element of the string splitted by "::"
+    */
+    public String extractItemType(String inputType) {
+        String[] splitted = inputType.split("::");
+        return splitted[splitted.length - 1];
+    }
+
+    /** 
+    * This method is used to extract the files's bit stream from an item.
+    * 
+    * @param DSpaceItem: The item to extract files from.
+    * @return The list of bit streams for the given item.
+    */
+    public List<Bitstream> extractItemFiles(Item DSpaceItem) {
+        for (Bundle bundle: DSpaceItem.getBundles()) {
+            if (bundle.getName().equals("ORIGINAL")) {
+                return bundle.getBitstreams();
+            }
+        }
+        return new ArrayList<Bitstream>();
+    }
+
+    /** 
+    * Converts a list of metadataValues into an HashMap for better data access.
+    * 
+    * @param metadataValues: The list of 'MetadataValue' to insert into the HashMap.
+    * @return The HashMap with all the values (key=fieldName, value= List of strings values).
+    */
+    public HashMap<String, List<String>> getValuesHashMap(List<MetadataValue> metadataValues){
+        HashMap<String, List<String>> hashMap = new HashMap<>();
+        for (MetadataValue metadataValue: metadataValues) {
+            String metadataField = metadataValue.getMetadataField().toString();
+            List<String> currentValueForFieldId = hashMap.get(metadataField);
```

Comment:
You could use something like 
```java
List<String> fieldList = hashMap.getOrDefault(metadataField, new ArrayList<String>());
fieldList.add(metadataValue.getMetadataField().toString());
```
[getOrDefault](https://docs.oracle.com/javase/8/docs/api/java/util/Hashtable.html#getOrDefault-java.lang.Object-V-)